### PR TITLE
Run as jenkins user and update pipeline library

### PR DESCRIPTION
**Pull request by hazim1093**
* Update base docker image version
* Run image as jenkins user to replace https://github.com/stakater-docker/maven-jenkins

### CODE REVIEW

**Comment by stakater-user** [comment]
@hazim1093 Yikes! You better fix it before anyone else finds out! [Build 1](https://jenkins.release.stakater.com/job/Stakater%20Docker/job/builder-maven/job/PR-4/1/) has Failed!

**Comment by stakater-user** [comment]
@hazim1093 Yikes! You better fix it before anyone else finds out! [Build 2](https://jenkins.release.stakater.com/job/Stakater%20Docker/job/builder-maven/job/PR-4/2/) has Failed!

**Comment by stakater-user** [comment]
@hazim1093 Image is available for testing. `docker pull docker.io/stakater/builder-maven:3.5.4-jdk1.8-apline8-SNAPSHOT-PR-6-14-SNAPSHOT-PR-4-3`

**Comment by stakater-user** [comment]
@hazim1093 Image is available for testing. `docker pull docker.io/stakater/builder-maven:3.5.4-jdk1.8-apline8-SNAPSHOT-PR-6-14-SNAPSHOT-PR-4-4`

**Comment by stakater-user** [comment]
@hazim1093 Image is available for testing. `docker pull docker.io/stakater/builder-maven:3.5.4-v1.16.8-SNAPSHOT-PR-4-5`

**Comment by stakater-user** [comment]
@hazim1093 Image is available for testing. `docker pull docker.io/stakater/builder-maven:3.5.4-v1.16.8-SNAPSHOT-PR-4-6`

**Comment by stakater-user** [comment]
@hazim1093 Image is available for testing. `docker pull docker.io/stakater/builder-maven:3.5.4-v1.16.8-SNAPSHOT-PR-4-7`

**Comment by stakater-user** [comment]
@hazim1093 Image is available for testing. `docker pull docker.io/stakater/builder-maven:3.5.4-jdk1.8-v1.16.8-SNAPSHOT-PR-4-8`

File: README.md [hunk 1]
```diff
@@ -1,2 +1,4 @@
 # builder-maven
 A maven builder image used to build maven apps
+
+This image runs as user `jenkins` with id `10000`, so that it can run maven as a non-root user which can access Jenkins directories and fiels as well, in order to seamlessly run as a Jenkins slave.
```
Comment (waseem-h) [inline README.md]: fiels should be files @hazim1093

**Comment by stakater-user** [comment]
@hazim1093 Yikes! You better fix it before anyone else finds out! [Build 9](https://jenkins.release.stakater.com/job/Stakater%20Docker/job/builder-maven/job/PR-4/9/) has Failed!

**Comment by stakater-user** [comment]
@hazim1093 Image is available for testing. `docker pull docker.io/stakater/builder-maven:3.5.4-jdk1.8-v1.16.8-SNAPSHOT-PR-4-10`

**Review by waseem-h — APPROVED** [review]

**Pull request merged**